cdc/sinkv2/eventsink/txn/worker.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package txn
+
+import (
+	"sync"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/pkg/chann"
+	"go.uber.org/zap"
+)
+
+type txnWithNotifier struct {
+	*txnEvent
+	wantMore func()
+}
+
+type worker struct {
+	ID      int
+	txnCh   *chann.Chann[txnWithNotifier]
+	stopped chan struct{}
+	wg      sync.WaitGroup
+	backend backend
+
+	// Fields only used in the background loop.
+	timer *time.Timer
+}
+
+func newWorker(ID int, backend backend) *worker {
+	return &worker{
+		ID:      ID,
+		txnCh:   chann.New[txnWithNotifier](chann.Cap(-1 /*unbounded*/)),
+		stopped: make(chan struct{}),
+		backend: backend,
+	}
+}
+
+func (w *worker) Add(txn *txnEvent, unlock func()) {
+	w.txnCh.In() <- txnWithNotifier{txn, unlock}
+}
+
+func (w *worker) Close() {
+	close(w.stopped)
+	w.wg.Wait()
+	w.txnCh.Close()
+}
+
+// Run a background loop.
+func (w *worker) Run() {
```

Comment:
One last suggestion: `Run()` method is usually blocking in the rest of the tiflow repo. It's better to keep it uniform.

Reply:
Or you can hide the method, and call it in the constructor.